lib/kb_SetUtilities/kb_SetUtilitiesImpl.py

```diff
@@ -58,24 +57,106 @@ def log(self, target, message):
         print(message)
         sys.stdout.flush()
 
-    def get_single_end_read_library(self, ws_data, ws_info, forward):
-        pass
-
-    def get_feature_set_seqs(self, ws_data, ws_info):
-        pass
-
-    def get_genome_feature_seqs(self, ws_data, ws_info):
-        pass
+    def check_params (self, params, required_params):
+        missing_params = []
+        for param in required_params:
+            if not params.get(param):
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            msg = []
+            for missing_param in missing_params:
+                msg.append('Missing required param: '+missing_param)
+                raise ValueError ("\n".join(msg))
```

Comment:
I'd put together a quick test for this to ensure that it is acting in the way that you expect.

It might be simpler just to do

```py
if len(missing_params):
    raise ValueError("Missing required param(s):\n" + "\n".join(missing_params))
```

Reply:
I really appreciate your thoroughness.  I think this code is simple enough that I don't want to take the extra time to add a test.  I have like 20 bigger priority things, including getting this PR merged so I can run the beta test Narrative in time to release this module at the App Release meeting at 1pm today.

Reply:
The code as-is won't work, which is why I suggested writing a test -- your code will throw an error on the first missing param. Switch in the line I suggested and it'll be fine.

Reply:
@ialarmedalien aha!  I indented the raise wrong!  I put in your code.